SpatialGDK/Source/SpatialGDK/Public/Schema/Ping.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/Component.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+namespace improbable
+{
+
+	struct ServerPing : Component
+	{
+		static const Worker_ComponentId ComponentId = SpatialConstants::SERVER_PING_COMPONENT_ID;
+
+		ServerPing() = default;
+		ServerPing(const Worker_ComponentData& Data)
+		{
+		}
+
+		FORCEINLINE Worker_ComponentData CreateServerPingData()
+		{
+			Worker_ComponentData Data = {};
+			Data.component_id = ComponentId;
+			Data.schema_type = Schema_CreateComponentData(ComponentId);
+
+			return Data;
+		}
+	};
+
+	struct ClientPong : Component
+	{
+		static const Worker_ComponentId ComponentId = SpatialConstants::CLIENT_PONG_COMPONENT_ID;
+
+		ClientPong() = default;
+		ClientPong(const Worker_ComponentData& Data)
+		{
+		}
+
+		FORCEINLINE Worker_ComponentData CreateClientPongData()
```

Comment:
Why FORCEINLINE?

Reply:
I remember linker complaining about functions like this if they aren't inline, and this probably doesn't need its own `.cpp` just for the one function.

Reply:
These have to be inline to be valid C++.  As otherwise they would violate ODR. However this probably doesn't need us to try and force the compiler to inline this. This will already have copy elision and I suspect this is short enough that the compiler will ignore any inline flags we give it.

Reply:
It's not wrong for it to be there. I'm just wondering why we want to add extra compiler hints.

Reply:
In fact this actually doesn't need an `inline` specifier as it's implicit if defined inside the struct definition.